pkg/services/oauthtoken/oauth_token.go

```diff
@@ -15,23 +19,28 @@ import (
 )
 
 var (
-	logger = log.New("oauthtoken")
+	logger      = log.New("oauthtoken")
+	ExpiryDelta = 10 * time.Second
 )
 
 type Service struct {
-	SocialService   social.Service
-	AuthInfoService login.AuthInfoService
+	SocialService     social.Service
+	AuthInfoService   login.AuthInfoService
+	singleFlightGroup *singleflight.Group
 }
 
 type OAuthTokenService interface {
 	GetCurrentOAuthToken(context.Context, *user.SignedInUser) *oauth2.Token
 	IsOAuthPassThruEnabled(*datasources.DataSource) bool
```

Comment:
Consider moving `IsOAuthPassThruEnabled` out of this interface because I think that is a concern of the DataProxy/PluginProxy and not tied tightly to handling the OAuth tokens.